lib/msf/core/data_store.rb

```diff
@@ -216,6 +219,22 @@ def from_file(path, name = 'global')
     end
   end
 
+  def merge!(other)
+    @options.merge!(other.options)
```

Comment:
Pro uses `#merge!` quite a bit with plain `Hash`es, so we will need to check if `other` is another `DataStore` or tag as `msf5`.